### PR TITLE
feat: add usage of segment in list

### PR DESCRIPTION
**Pull request by sjaanus**
image

### CODE REVIEW

File: src/lib/types/model.ts [hunk 1]
```diff
@@ -402,6 +402,8 @@ export interface ISegment {
     description?: string;
     project?: string;
     constraints: IConstraint[];
+    usedInProjects?: number;
```
Comment (kwasniew) [inline src/lib/types/model.ts:405]: does it have to be optional? Is there any scenario where we don't know the number?

File: src/lib/db/segment-store.ts [hunk 2]
```diff
@@ -97,8 +99,29 @@ export default class SegmentStore implements ISegmentStore {
 
     async getAll(): Promise<ISegment[]> {
         const rows: ISegmentRow[] = await this.db
-            .select(this.prefixColumns())
+            .select(
+                this.prefixColumns(),
+                'used_in_projects',
+                'used_in_features',
+            )
+            .countDistinct(
```
Comment (kwasniew) [inline src/lib/db/segment-store.ts:107]: have you tested it with a big data set in terms of performance?

File: frontend/src/component/segments/SegmentTable.tsx [hunk 3]
```diff
@@ -182,8 +182,34 @@ const COLUMNS = [
...
+    },
+    {
+        Header: 'Used in',
+        width: '60%',
         Cell: ({ value, row: { original } }: any) => (
-            <HighlightCell value={value} subtitle={original.description} />
+            <TextCell
+                sx={{
+                    color:
+                        original.usedInProjects == 0 &&
```
Comment (kwasniew) [inline frontend/src/component/segments/SegmentTable.tsx:204]: === as GH actions bots says

File: src/lib/db/segment-store.ts [hunk 4]
```diff
@@ -97,8 +99,29 @@ export default class SegmentStore implements ISegmentStore {
 
     async getAll(): Promise<ISegment[]> {
         const rows: ISegmentRow[] = await this.db
-            .select(this.prefixColumns())
+            .select(
```
Comment (kwasniew) [inline src/lib/db/segment-store.ts:102]: do we have any backend test that will tell us when this query breaks?

File: frontend/src/component/segments/SegmentTable.tsx [hunk 5]
```diff
@@ -182,8 +182,34 @@ const COLUMNS = [
         Header: 'Name',
         accessor: 'name',
         width: '60%',
+        Cell: ({
```
Comment (kwasniew) [inline frontend/src/component/segments/SegmentTable.tsx:185]: I'd enhance existing test for a segmenttable or create a new one to test enabled and disabled color display logic

**Review by kwasniew — CHANGES_REQUESTED** [review]

File: frontend/src/component/segments/SegmentTable.tsx (hunk 5 above)
Comment (kwasniew) [reply frontend/src/component/segments/SegmentTable.tsx:185]: can be in a separate PR

File: src/lib/db/segment-store.ts (hunk 2 above)
Comment (sjaanus) [reply src/lib/db/segment-store.ts:107]: Analyzed this query on enterprise, it is just regular join, quite cheap.

File: src/lib/types/model.ts (hunk 1 above)
Comment (sjaanus) [reply src/lib/types/model.ts:405]: Yes, only case when we know the number is the when we load the list. Throughout app, other places, we dont know it about segment

**Review by kwasniew — APPROVED** [review]

File: frontend/src/component/segments/SegmentTable.tsx [hunk 6]
```diff
@@ -56,6 +65,10 @@ export const SegmentTable = () => {
         return segments;
     }, [segments, projectId]);
 
+    const columns = useMemo(
```
Comment (sjaanus) [inline frontend/src/component/segments/SegmentTable.tsx:68]: Is this part fine?

**Pull request merged**